magento/magento2
# Grouped product frontend quantity validation added and code refactor

### Description (*)
Grouped Product Frontend Quantity validation added

### Related Pull Requests


### Fixed Issues (if relevant)
1. #39479 
2. #39142

### Manual testing scenarios (*)


1.Create a Grouped Product and associate it with child products.
2. Set the following quantity properties for one of the child products:
    - Minimum Qty Allowed in Shopping Cart: 2
    - Maximum Qty Allowed in Shopping Cart: 10
    - Qty Increments: 2
 3.Open the Product Detail Page (PDP) of the Grouped Product and test invalid quantities for the selected child product (e.g., 1, 3, or 11) to confirm that invalid inputs are restricted.



### Contribution checklist (*)
